java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java

```diff
@@ -2357,7 +2357,7 @@ public void testDecimalVectorEquals() {
       VectorEqualsVisitor visitor2 = new VectorEqualsVisitor();
 
       assertTrue(visitor1.vectorEquals(vector1, vector2));
-      assertFalse(visitor2.vectorEquals(vector1, vector3));
+      assertTrue(visitor2.vectorEquals(vector1, vector3));
```

Comment:
I'm not sure it is correct to flip since the scale is different between the two vectors.  @pravindra ?

Reply:
This is a general question. 
A numeric can be represented as different decimal values, with different scales and precisions. 

The question is: is the same numeric value equal when represented in different format?

For BigDecimal in standard java, there are two ways to compare:
1. The equals method compares both value and scale.
2. The compareTo method comapres the underlying value.

From a performance view point, comparing by the underlying value is more performant for Arrow's memory layout?

Reply:
My impression is the comparison is actually  .1 == 1 and .2 == 2 (i.e not the same as java BigDecimal, I might be misreading it but if you look at BigDecimal.getObject, I think this agrees with my assessment).

Reply:
@emkornfield You are right.
I am wrong because Arrow is different from BigDecimal. Arrow stores unscaled values.
Sorry for my mistake.

I have updated the code to revert this flip. Please take a look.

Reply:
@emkornfield After more thoughts, I think UnionVector is a special case: for union vectors, it is possible that two vectors have different fields, but have equal values in a certain range. 

So the changes for the RangeEqualsVisitor are like this: for union vectors, we only compare minor types, for other vectors, we compare fields. 

The problem for DecimalVector no longer exisits.